Common/Securities/Option/OptionFilterUniverse.cs

```diff
@@ -853,9 +1029,22 @@ public static class OptionFilterUniverseEx
         /// <param name="universe">Universe to apply the filter too</param>
         /// <param name="predicate">Bool function to determine which Symbol are filtered</param>
         /// <returns>Universe with filter applied</returns>
-        public static OptionFilterUniverse Where(this OptionFilterUniverse universe, Func<Symbol, bool> predicate)
+        public static OptionFilterUniverse Where(this OptionFilterUniverse universe, Func<OptionUniverse, bool> predicate)
         {
-            universe.AllSymbols = universe.AllSymbols.Where(predicate).ToList();
+            universe.AllSymbols = universe.Data.Where(predicate).Select(x => x.Symbol).ToList();
+            return universe;
+        }
+
+        /// <summary>
+        /// Filters universe
+        /// </summary>
+        /// <param name="universe">Universe to apply the filter too</param>
+        /// <param name="predicate">Bool function to determine which Symbol are filtered</param>
+        /// <returns>Universe with filter applied</returns>
+        public static OptionFilterUniverse Where(this OptionFilterUniverse universe, PyObject predicate)
+        {
+            using var _ = Py.GIL();
```

Comment:
Don't think we need the Gil here, same below, since `ConvertToDelegate` internally takes it already?